manager/utils.go

```diff
@@ -425,3 +425,123 @@ func GetTracepointID(category, name string) (int, error) {
 	}
 	return tracepointID, nil
 }
+
+// KProbePMUType is used to cache the KProbe PMU type
+var KProbePMUType uint32
+
+// UProbePMUType is used to cache the UProbe PMU type
+var UProbePMUType uint32
+
+// FindPMUType returns the PMU type of the provided event type
+// This function tries to use /sys/bus/event_source/devices/%s/type.
+func FindPMUType(eventType string) (uint32, error) {
+	switch eventType {
+	case "kprobe", "kretprobe":
+		if KProbePMUType != 0 {
+			return KProbePMUType, nil
+		}
+		if eventType == "kretprobe" {
+			eventType = "kprobe"
+		}
+	case "uprobe", "uretprobe":
+		if UProbePMUType != 0 {
+			return UProbePMUType, nil
+		}
+		if eventType == "uretprobe" {
+			eventType = "uprobe"
+		}
+	default:
+		return 0, errors.Errorf("no PMU type for %s", eventType)
+	}
+
+	PMUTypeFile := fmt.Sprintf("/sys/bus/event_source/devices/%s/type", eventType)
+	typeBytes, err := ioutil.ReadFile(PMUTypeFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			PMUType, err := strconv.Atoi(strings.TrimSpace(string(typeBytes)))
+			if err != nil {
+				return 0, err
+			}
+
+			switch eventType {
+			case "kprobe":
+				KProbePMUType = uint32(PMUType)
+			case "uprobe":
+				UProbePMUType = uint32(PMUType)
+			}
+			return uint32(PMUType), nil
+		}
+	}
+
+	// sysfs might not be mounted, return default values
```

Comment:
Is there a danger of using default values here? Could we end up creating incorrect probes?

Reply:
From what I've seen in the kernel, it would require that a PMU device is manually added in the list of PMU devices before the kprobe & uprobe ones.
I initially added it because I wanted to see if having default values could help us get rid of system-probe's AppArmor profile (no need for r/w access to debugfs anymore).

Anyway, getting rid of it because we don't really need it right now.